### DISCUSSION TITLE
🛣️ Roadmap: v1.0.0-v2.0.0

### DISCUSSION THREAD

**Discussion by blefnk** [blefnk/relivator#13] [General]
## Roadmap Checklist For `v2.0.0` Release

This roadmap outlines the key features and improvements to be implemented in this Next.js starter. Not marked things may be already configured, but was not tested too much. If mistakes here, please create an issue.

1. [x] ✅ [Release v0.0.0-v1.0.0 versions](https://github.com/blefnk/relivator/releases/tag/1.0.0) of the Relivator.
2. [ ] 🏡 A completely new landing page with a unique design and revamped components, along with a full redesign of all other pages and their components.
3. [ ] 🦾 Partially complete [BA11YC (Bleverse Accessability Convention) checklist](https://github.com/bs-oss/BA11YC).
4. [ ] 🧁 Full [Bun](https://bun.sh) support and compatibility.
5. [ ] 💎 Advanced implementation of user subscriptions and checkout system via **Stripe**.
6. [ ] 🏬 Advanced implementation of storefront with products, categories, and subcategories.
7. [ ] 🤖 Implement AI features and chat, using e.g [Vercel AI SDK](https://sdk.vercel.ai/docs), to this starter (read: [Introducing the Vercel AI SDK](https://vercel.com/blog/introducing-the-vercel-ai-sdk)).
8. [ ] 🌩️ Full Next.js Edge things support and compatibility.
9. [ ] 😇 Maximum accessability **for users**. Accessability **for developers**, both beginners and seniors. Accessability **for bots**, like [Googlebot](https://developers.google.com/search/docs/crawling-indexing/googlebot) or [PageSpeed Insights Crawler](https://pagespeed.web.dev/). Accessability **for everyone**.
10. [ ] 😺 Using some interesting recommendations from \*\*[Material Design 3](https://m3.material.io) and other design systems.
11. [ ] 📝 Comment system for products, in types of Review and Question.
12. [ ] 🔔 Advanced notification system, like toaster, pop up, and page.
13. [ ] 💠 Advanced 404 Not Found page with full internationalization support.
14. [ ] 🪪 Advanced sign-up/sign-in/restoring by using email-password and magic-links.
15. [ ] 🏎️ Try to optimize everything in the app to make also faster dev cold start and build.
16. [ ] 🪄 Use [Kysely](https://kysely.dev/) with Drizzle to get the full type-safety for TypeScript SQL query builder.
17. [ ] 🔀 Implement some useful tests, including if possible stress-tests to simulate and check app performance, when many people uses the app at one time.
18. [ ] 🔗 Minimal how it's possible HTML tags nesting. And also correct ways of using HTML tags.
19. [ ] ⚙️ Beautifully configured `app.ts`, a single config to replace them all.
20. [ ] ✒️ Gorgeously written comments everywhere, keeping the code still clean.
21. [ ] 🐎 Nicely configured `package.json` with correctly installed and sorted packages in `dependencies` and `devDependencies`.
22. [ ] ✨ Admin dashboard with stores, products, orders, subscriptions, and payments.
23. [ ] ♨️ Advanced indicators implementations for installed packages, environment variables, and improvements to TailwindCSS screen sizes.
24. [ ] 🦄 Project is clean from duplicates, like files, components, etc.
25. [ ] 🌬️ Advanced theme switching without flashing, using Tailwind Dark Mode with [React Server Side support](https://michaelangelo.io/blog/darkmode-rsc) and dynamic cookies.
26. [ ] ⬆️ File uploads implementation using [UploadThing](https://uploadthing.com) and [Cloudinary](https://cloudinary.com).
27. [ ] 🏟️ Everything in the project is sorted in ascending order, unless a different sort is required somewhere.
28. [ ] 🌀 Comprehensive configured Next.js 13 App Router, API routes handled by Route Handlers, with the RSCs and everything all new.
29. [ ] 💫 Use where it make sense the [next-usequerystate](https://github.com/47ng/next-usequerystate) library ([read the article](https://francoisbest.com/posts/2023/storing-react-state-in-the-url-with-nextjs)).
30. [ ] 🌅 Using every usable [TanStack](https://tanstack.com) libraries.
31. [ ] 🌌 Even more comprehensive i18n implementation, with using country and locale codes, supporting 20 languages. Every language must be verified by native speakers after machine translation.
32. [ ] 🌞 Clean from any of unused things, like packages, libs, vars, etc.
33. [ ] 🍪 Minimal possible number of cookies, to be ready for cookie-free future, with cookie notification implementation.
34. [ ] ⭐ Complete advanced sign up and sign in pages both with socials and classic form methods.
35. [ ] 🐧 The most beginner-friendly and prettyful starter, how it's possible.
36. [ ] 🦊 Every possible page is wrapped using predefined shell wrappers.
37. [ ] ❄️ Project is clean from `@ts-expect-error` or similar things.
38. [ ] 🐭 Minimal number of config files, how it possible.
39. [ ] 🌳 Using `function` instead `const` for components everywhere, to make it more cleaner and beginners-friendly.
40. [ ] 🦑 Absolute paths are used wherever it makes sense.
41. [ ] 🐳 [Jest](https://jestjs.io) testing, optimized for Next.js 13.
42. [ ] 📰 Email verification and newsletter subscription and email marketing via [Resend](https://resend.com) and [React Email](https://react.email).
43. [ ] 🤖 Ultimate type-safety using strict-mode in [TypeScript](https://typescriptlang.org), typedRoutes, zod, middlewares etc.
44. [ ] 🪟 Pop-ups like: cookies/GDPR notification (with corresponding management settings page), Google floating notification about fast login, etc.
45. [ ] 🤝 Seller and customer workflows.
46. [ ] 🦾 Accessability everywhere, for app users UI (User Interface) with UX (User Experience), and for developers DX (Developer Experience). And with usability, but still with beauty in the mind.
47. [ ] 🌉 Better `middleware.ts` configuration with multi-mw implementation.
48. [ ] 🪶 All website languages must be free of grammatical and spelling errors and written in accordance with the newest rules of each language.
49. [ ] 😜 Minimal possible references to Relivator, Bleverse, etc. to be really easy-starter. Using variables from `app.ts` everywhere.
50. [ ] 🎯 Maximum possible for the app performance points on services like Google PageSpeed Insights. Also app must pass all hard tests.

**Comment by bmason** [comment]
Ambitious.  You have done good work.

**Comment by demiroo** [comment]
i cant wait
